backend/src/main/java/com/google/sps/servlets/SearchMatchServlet.java

```diff
@@ -92,14 +97,59 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     participantDatastore.removeParticipant(match.getFirstParticipantUsername());
     participantDatastore.removeParticipant(match.getSecondParticipantUsername());
 
+    sendMatchResponse(response, match);
+  }
+
+  /** Return true if expired and should be removed, false if still valid */
+  private boolean checkExpired(Participant participant) {
+    long currentTimeMillis = System.currentTimeMillis();
+    return (participant.getEndTimeAvailable()
+            - TimeUnit.MINUTES.toMillis(participant.getDuration() + PADDING_MINUTES))
+        < currentTimeMillis;
+  }
+
+  /** Send JSON response for expired participant that has been removed from datastore */
+  private void sendExpiredResponse(HttpServletResponse response) {
+    JSONObject expired = new JSONObject();
+    expired.put(JSON_MATCH_STATUS, "expired");
+
+    // Send the JSON back as the response
+    response.setContentType("application/json");
+    try {
+      response.getWriter().println(expired.toString());
+    } catch (IOException e) {
+      System.out.println("Failed to return expired participant response.");
+    }
```

Comment:
Before this refactor doGet would throw these IOExceptions but now they'll be swallowed in these methods. Was that a deliberate change or can we change the methods signatures to include `throws IOException` and remove the catches?

Reply:
+1. Maybe we can throw the exceptions with the error messages you are outputting